src/main/export/export.ts

```diff
@@ -1,53 +1,71 @@
 import { BrowserWindow } from 'electron';
-import { writeFileSync } from 'fs';
+import { existsSync, writeFileSync } from 'fs';
 import path, { join } from 'path';
-import { secondToTimestamp, padZeros } from '../timeUtils';
+import ffprobe from 'ffprobe';
+import ffprobeStatic from 'ffprobe-static';
+import { secondToEDLTimestamp, padZeros } from '../timeUtils';
 import { RuntimeProject, Transcription } from '../../sharedTypes';
 import { mkdir } from '../util';
 import convertTranscriptToCuts from '../../transcriptProcessing/transcriptToCuts';
+import { fracFpsToDec } from '../handlers/helpers/exportUtils';
 
 export const constructEDL: (
   title: string,
   transcription: Transcription,
   source: string | null,
   mainWindow: BrowserWindow | null
-) => string = (title, transcription, source, mainWindow) => {
-  let output = `TITLE: ${title}\nFCM: NON-DROP FRAME\n\n`;
+) => Promise<string> = async (title, transcription, source, mainWindow) => {
+  if (!source) {
+    throw Error('No Video Source');
+  } else {
+    let fps = 30; // default to a safe 30fps.
+    if (existsSync(source)) {
+      const videoData = await ffprobe(source, { path: ffprobeStatic.path });
+      fps = fracFpsToDec(videoData.streams[0].avg_frame_rate);
+    }
 
-  const cuts = convertTranscriptToCuts(transcription);
-  const entries = cuts.length;
+    let output = `TITLE: ${title}\nFCM: NON-DROP FRAME\n\n`;
+    // this can be technically incorrect in some rare cases, but it doesn't affect functionality
 
-  const timeline = {
-    start: 0,
-    end: 0,
-  };
+    const cuts = convertTranscriptToCuts(transcription);
+    const entries = cuts.length;
 
-  output += cuts
-    .map((cut, i) => {
-      const edlEntry = `${padZeros(
-        i + 1,
-        Math.max(Math.floor(Math.log10(entries)) + 1, 3)
-      )}  AX       AA/V  C`;
+    const timeline = {
+      start: 0,
+      end: 0,
+    };
 
-      const editStart = secondToTimestamp(cut.startTime);
-      const editEnd = secondToTimestamp(cut.startTime + cut.duration);
+    output += cuts
+      .map((cut, i) => {
+        const edlEntry = `${padZeros(
+          i + 1,
+          Math.max(Math.floor(Math.log10(entries)) + 1, 3)
+        )}  AX       AA/V  C`;
 
-      timeline.start = timeline.end;
-      timeline.end = timeline.start + cut.duration;
+        const editStart = secondToEDLTimestamp(cut.startTime, fps);
+        const editEnd = secondToEDLTimestamp(
+          cut.startTime + cut.duration + 1 / fps,
+          fps
+        );
+        /**
+         * Addressed the gap problem by rounding up 1 frame per cut, it will be overwritten in case there is no rounding error,
+         * so this will only work to fill in microgaps. LMK if a less jank solution is wanted.
+         */
 
-      const timelineStart = secondToTimestamp(timeline.start);
-      const timelineEnd = secondToTimestamp(timeline.end);
+        timeline.start = timeline.end;
+        timeline.end = timeline.start + cut.duration;
```

Comment:
Since `timeline.start` is equal to `timeline.end` due to the line before, can just replace this with `timeline.end += cut.duration`

Reply:
Implemented! It was originally like that for readability, but I agree that doing it with `+=` doesn't make it unclear